osbuild/util/path.py

```diff
@@ -1,5 +1,39 @@
 """Path handling utility functions"""
+import errno
 import os
+import os.path
+
+from .ctx import suppress_oserror
+
+
+def clamp_mtime(path: str, start: int, to: int):
+    """Clamp all modification times of 'path'
+
+    Set the mtime of 'path' to 'to' if it is greater or equal to 'start'.
+    If 'to' is None, the mtime is set to the current time.
+    """
+
+    times = (to, to)
+
+    def fix_utime(path, dfd: int):
+        sb = os.stat(path, dir_fd=dfd, follow_symlinks=False)
+        if sb.st_mtime < start:
+            return
+
+        # We might get a permission error when the immutable flag is set;
+        # since there is nothing much we can do, we just ignore it
```

Comment:
We could lift the immutable flag temporarily?

Reply:
like a:
```
with lift_immutable(path):
   ...
```

neat idea! I like it

Reply:
I think I will leave this for a follow up.